earthaccess/search.py

```diff
@@ -1,16 +1,62 @@
 import datetime as dt
 from inspect import getmembers, ismethod
-from typing import Any, Dict, List, Optional, Tuple, Type, Union
+from typing import Any, List, Optional, Tuple, Type, Union
 
 import dateutil.parser as parser  # type: ignore
-from cmr import CollectionQuery, GranuleQuery  # type: ignore
-from requests import exceptions, session
+import requests
+from cmr import CollectionQuery, GranuleQuery
 
+# type: ignore
 from .auth import Auth
 from .daac import find_provider, find_provider_by_shortname
 from .results import DataCollection, DataGranule
 
 
+def get_results(request, limit: int = 2000) -> list:  # type: ignore
+    """
+    Get all results up to some limit, even if spanning multiple pages.
+
+    ???+ Tip
+        The default page size is 2000, if the supplied value is greater then the Search-After header
+        will be used to iterate across multiple requests until either the limit has been reached
+        or there are no more results.
+    Parameters:
+        limit: The number of results to return
+
+    Returns:
+        query results as a list
+    """
+
+    page_size = min(limit, 2000)
+    url = request._build_url()
+
+    results: List = []
+    more_results = True
+    while more_results:
+        # Only get what we need
+        page_size = min(limit - len(results), page_size)
+        response = requests.get(
+            url, headers=request.headers, params={"page_size": page_size}
+        )
+        if request.headers is None:
+            request.headers = {}
+        request.headers["cmr-search-after"] = response.headers["cmr-search-after"]
+
+        try:
+            response.raise_for_status()
+        except requests.exceptions.HTTPError as ex:
+            raise RuntimeError(ex.response.text)
+
+        latest = response.json()["items"]
```

Comment:
This works only when the format is `"umm_json"`. Your refactoring dropped the logic for format `"json"` along with the "catch-all" case where the format is neither `"umm_json"` nor `"json"`.

Since dropping that logic did not cause unit tests to fail, this indicates a gap in the unit test suite that I'd suggest you fill as part of this PR.